grype/matcher/common/utils_test.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/go-test/deep"
+	"github.com/scylladb/go-set"
+	"github.com/stretchr/testify/assert"
+)
+
+func assertMatchesWithoutVulnData(t testing.TB, expected, actual []match.Match) {
```

Comment:
This helper function looks really useful. It took me a while to figure out what it does and how (still chewing on it actually). I have a couple of thoughts on how to improve that, but I'm open to discussion/suggestions!

First, it looks like this is comparing matches, including a single field of vulnerability data: ID. It would be ideal to find a name that reflects that. Perhaps something like `assertMatchesUsingIDsForVulnerabilities`. I don't love it, but the idea is to describe how vuln data is being compared within the matches (e.g., instead of signaling that vuln data is being disregarded altogether in the assertion).

Second, the implementation is taking me a while to wrap my head around. One question I still have is: what's being caught in the stringset comparison that's not caught in the earlier diffing? Another question I have is: does the order of the actual slice of matches matter? Looking at the `for` loop, my understanding is leaning toward yes. If that's true, would something like this work as the function body? (And if this is insufficient, it'd help to better understand why)

```go
for idx, a := range actual {
	// the vulnerability comparison should use only the vulnerability ID
	a.Vulnerability = vulnerability.Vulnerability{ID: a.Vulnerability.ID}
	assert.Equal(t, expected[idx], a)
}
```

Reply:
this makes sense --except for the deep.Equal replacement, I'll adjust the PR. (I found assert.Equal and deep.Equal were behaving a slightly differently in this context, didn't investigate any further for now)